node/resolvers/auth/index.ts

```diff
@@ -14,15 +16,24 @@ const makeRequest = async (ctx, url) => {
 }
 
 export const mutations = {
-  sendEmailVerification: async (_, args, { vtex: ioContext }) => {
-    const { data: { authenticationToken } } = await makeRequest(ioContext, paths.getTemporaryToken())
-    await makeRequest(ioContext, paths.sendEmailVerification(args.email, authenticationToken))
-    return { authToken: authenticationToken }
+  sendEmailVerification: async (_, args, { vtex: ioContext, response }) => {
+    const { data, status } = await makeRequest(ioContext, paths.getTemporaryToken(ioContext.account, ioContext.account))
+    if (!data.authenticationToken) {
+      throw new ResolverError(`ERROR ${data}`, status)
+    }
+    await makeRequest(ioContext, paths.sendEmailVerification(args.email, data.authenticationToken))
+    response.set('Set-Cookie', serialize('temporarySession', data.authenticationToken, { path: '/', }))
+    return data.authenticationToken ? true : false
```

Comment:
The `if (!data.authenticationToken)` in line 21 already checks if data.authenticationToken has a valid value. You could just return true here.